sync-core/src/test/java/com/cloudant/http/HttpTest.java

```diff
@@ -119,4 +122,52 @@ public void testReadBeforeExecute() throws IOException {
         Assert.assertEquals(bis.available(), data.getBytes().length);
         client.deleteDb();
     }
+
+
+    //NOTE: This test doesn't work with specified couch servers,
```

Comment:
Due to the way the specified couch servers inject credentials, i.e.  through the URI, the tests will always pass even if the cookie filter failed to get a cookie.